web/vtadmin/src/components/routes/keyspaces/KeyspaceActions.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from 'react';
+import Dropdown from '../../dropdown/Dropdown';
+import MenuItem from '../../dropdown/MenuItem';
+import { Icons } from '../../Icon';
+import Toggle from '../../toggle/Toggle';
+import { useValidateKeyspace, useValidateSchemaKeyspace, useValidateVersionKeyspace } from '../../../hooks/api';
+import KeyspaceAction from './KeyspaceAction';
+
+interface KeyspaceActionsProps {
+    keyspace: string;
+    clusterID: string;
+}
+
+const KeyspaceActions: React.FC<KeyspaceActionsProps> = ({ keyspace, clusterID }) => {
+    const [currentDialog, setCurrentDialog] = useState<string | null>(null);
```

Comment:
Since we are both reading + writing these string values from a fixed set, I think a string enum would be helpful here. It'd give us a lil more type safety + also prevent typos.